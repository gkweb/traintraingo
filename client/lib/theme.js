import React, {Component} from 'react'

const themes = {
  light: {
    'name': 'light',
    'values': {
      'primaryBg': '#f7f7f7',
      'secondaryBg': '#ddd',
      'tertiaryBg': '#bbb',
      'inversePrimaryBg': '#f7f7f7',
      'primary': '#4c4c4c',
      'secondary': '#9a9a9a',
      'tertiary': '#f7f7f7',
      'inversePrimary': '#4c4c4c',
      'highlightPrimary': '#EEDB57',
      'highlightSecondary': '#ffB86f',
      'highlightTertiary': '',
      'logoPrimaryBg': '#5EC64C',
      'logoSecondaryBg': '#105C26',
      'logoPrimary': '#4c4c4c',
      'logoSecondary': '#fff',
      'input': '#222',
      'inputBg': '#fff',
      'inputBorder': '#ccc',
      'inputIcon': 'rgb(76, 76, 76, .75)',
      'link': '#4495d4',
      'linkBg': '#f7f7f7',
      'linkHoverBg': '#fff'
    }
  },
  dark: {
    'name': 'dark',
    'values': {
      'primaryBg': '#444',
      'secondaryBg': '#333',
      'tertiaryBg': '#222',
      'inversePrimaryBg': '#fff',
      'primary': '#fff',
      'secondary': '#ddd',
      'tertiary': '#000',
      'inversePrimary': '#000',
      'highlightPrimary': '#EEDB57',
      'highlightSecondary': '#ffB86f',
      'highlightTertiary': '',
      'logoPrimaryBg': '#5EC64C',
      'logoSecondaryBg': '#105C26',
      'logoPrimary': '#fff',
      'logoSecondary': '#fff',
      'input': '#fff',
      'inputBg': '#222',
      'inputBorder': '#000',
      'inputIcon': 'rgba(255, 255, 255, .25)',
      'link': '#fff',
      'linkBg': '#333',
      'linkHoverBg': '#222'
    }
  }
}

const ThemeManagementContext = React.createContext({
    activeTheme: themes.light.name,
    theme: themes.light // default value
  }
)

export {
  themes,
  ThemeManagementContext
}
