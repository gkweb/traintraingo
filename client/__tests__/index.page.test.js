import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import { themes } from './../lib/theme'
import { create } from 'react-test-renderer'
import Index from '../pages/index'
import 'jest-styled-components'

test('Logo has Theme Context vals passed down to it', () => {
  const c = mount(
    <ThemeProvider theme={themes.light.values}>
      <Index />
    </ThemeProvider>
  )

  expect(c.find('circle[fill="#105C26"]').exists()).toBe(true)
})

test('Home / Index Snapshot', () => {
  const c = shallow(<Index />, { theme: themes.light.values })
  expect(expect(c).toMatchSnapshot()).toMatchSnapshot()
})
