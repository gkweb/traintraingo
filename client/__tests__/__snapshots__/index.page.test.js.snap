// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home / Index Snapshot 1`] = `
<styled.div>
  <b
    config={
      Object {
        "description": "Train train go",
        "title": "Train Train Go ðŸš†ðŸš†ðŸš¶ - Train departures without the bloat.",
      }
    }
  />
  <Styled(styled.main)>
    <styled.div>
      <Menu />
    </styled.div>
    <styled.div>
      <styled.div>
        <WithTheme(Logo) />
      </styled.div>
      <styled.div>
        <Search
          initialRun={true}
          onChange={[Function]}
          onSubmit={[Function]}
        />
        <styled.p>
          Melbourne, Australia PTV train departures with no bloat.
        </styled.p>
      </styled.div>
    </styled.div>
    <Favourites />
  </Styled(styled.main)>
</styled.div>
`;

exports[`Home / Index Snapshot 2`] = `undefined`;
